### ISSUE TITLE
TrinityCore: some worldserver.conf fields ignored

### ISSUE THREAD

**Issue by ihm-tswow** [tswow/tswow#405]
from user:

Expansion = 0 works
CharacterCreating.Disabled.RaceMask = 1536 does not work
CharacterCreating.Disabled.ClassMask = 32 does not work
DungeonFinder.OptionsMask = 0 does not work

- Need to reproduce
- Need to see if this is a tswow or trinitycore issue

**Comment by kaytotes** [comment]
@ihm-tswow I have tested this with a TC built fresh about an hour ago and gone over this in detail. Appears to me that the default TrinityCore behaviour for all 3 of these has changed. 

`RaceMask` used to make the race not selectable and greyed out. The same with `ClassMask`.

Now the check happens when you go to create the character and pops up this. 

![image](https://user-images.githubusercontent.com/7526918/146662992-ec53eac9-a3ab-4bdc-b8dc-afa21b43af94.png)

GM accounts can make them no matter what so I tested on those and normal accounts. 

`DungeonFinder.OptionsMask` used to grey out the Dungeon Finder button but now instead simply silently does nothing with this set to 0. User never joins the queue but no indication why. With it set to 1 joining is all fine. 

So, to summarize, there is no bug here. TC just changed the behaviour. 

Preventing these races from being selected is perfectly in the bounds of a simple datascript in that case so I'll close this and solve it in my project.

**Comment by ihm-tswow** [comment]
Good to know, thanks for the report!
